projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -9,20 +9,46 @@
  * Class Jetpack_Podcast_Helper
  */
 class Jetpack_Podcast_Helper {
+	const MIN_PODCAST_CACHE_TIMEOUT = 30 * MINUTE_IN_SECONDS;
+
 	/**
 	 * The RSS feed of the podcast.
 	 *
 	 * @var string
 	 */
 	protected $feed = null;
 
+	/**
+	 * The number of seconds to cache the podcast feed data.
+	 * This should be specified using the `jetpack_podcast_feed_cache_timeout` filter,
+	 * and must be greater than {@see self::MIN_PODCAST_CACHE_TIMEOUT}.
+	 *
+	 * @var int|null
+	 */
+	protected $cache_timeout = null;
+
 	/**
 	 * Initialize class.
 	 *
 	 * @param string $feed The RSS feed of the podcast.
 	 */
 	public function __construct( $feed ) {
 		$this->feed = esc_url_raw( $feed );
+
+		/**
+		 * Filter the number of seconds to cache a specific podcast URL for. The returned value will be ignored if it is not a valid integer.
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @param string   $podcast_url   The URL of the podcast feed.
+		 * @param int|null $cache_timeout The number of seconds to cache the podcast data. Default value is null.
+		 */
+		$podcast_cache_timeout = apply_filters( 'jetpack_podcast_feed_cache_timeout', $this->feed, null );
```

Comment:
The default value in this case is `null` - we're using that to indicate we should fall through to the default value returned from the `wp_feed_cache_transient_lifetime` filter. We could definitely invert this to apply that filter to get our default value, but that feels like it could easily get into recursive territory if we then hook into the same filter to inject our custom value. Maybe this needs additional documentation to clarify why this is unusual?

I would be more than happy to change, but I wanted to try and do less if we don't have a custom value supplied by the environment.

Reply:
In that case, with `null` as the default timeout, I would expect to see the following, with `null` as the second argument.
```
apply_filters( 'jetpack_podcast_feed_cache_timeout', null, $this->feed );
```
Because the timeout is what's being filtered, not the feed url.

Reply:
🤦 You're 100% right!

I've updated this in 1c5d462 to switch the argument order (and documentation), as well as using the default value instead of `null`.